presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -125,15 +191,23 @@ public RetryDriver withClassifier(Function<Exception, Exception> classifier)
 
         while (true) {
             retryStatus.nextAttempt();
+            getStats(callableName, stats -> stats.updateAttempt());
             try {
-                return callable.call();
+                V result = callable.call();
+                getStats(callableName, stats -> stats.updateSuccess());
+                getStats(callableName, stats -> stats.updateSucceedAttempts(retryStatus.attempts.get()));
+                getStats(callableName, stats -> stats.updateSucceedLatency(retryStatus.getDuration()));
+                return result;
             }
             catch (InterruptedException ie) {
                 Thread.currentThread().interrupt();
                 throw propagate(ie);
             }
             catch (Exception e) {
                 if (retryStatus.shouldStopRetry(e)) {
+                    getStats(callableName, stats -> stats.updateFinalFailure());
```

Comment:
Let's record failure stats in `propagate` in order not to miss some of them, for example those caused by interrupt.